lib/srv/regular/sftp.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package regular
+
+import (
+	"context"
+	"io"
+	"os"
+	"os/exec"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/lib/srv"
+	"github.com/gravitational/trace"
+
+	"github.com/sirupsen/logrus"
+	"golang.org/x/crypto/ssh"
+)
+
+type sftpSubsys struct {
+	sftpCmd *exec.Cmd
+	errCh   chan error
+	log     *logrus.Entry
+}
+
+func newSftpSubsys() (*sftpSubsys, error) {
+	// TODO: add prometheus collectors?
+	return &sftpSubsys{
+		log: logrus.WithFields(logrus.Fields{
+			trace.Component:       teleport.ComponentSubsystemSFTP,
+			trace.ComponentFields: map[string]string{},
+		}),
+	}, nil
+}
+
+func (s *sftpSubsys) Start(ctx context.Context, _ *ssh.ServerConn, ch ssh.Channel, req *ssh.Request, serverCtx *srv.ServerContext) error {
+	err := req.Reply(true, nil)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	chrr, chrw, err := os.Pipe()
```

Comment:
These variable names are so close together it's confusing me a bit. Any ideas for better names?